python/python/lance/dataset.py

```diff
@@ -169,9 +169,8 @@ def take(self, indices, **kwargs):
         -------
         table : Table
         """
-        # TODO expose take from Rust
         # kwargs['take'] = indices
-        return self.scanner(**kwargs).to_table().take(indices)
+        return self._ds.take(indices)
```

Comment:
This is nice!!

Could you add a unit test to verify that the returned object is `Table` instead of `RecordBatch`? 

https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Dataset.html#pyarrow.dataset.Dataset.take